src/sensor/parser-ping.cpp

```diff
@@ -0,0 +1,34 @@
+#include "parser-ping.h"
+
+void PingParserExt::clearBuffer()
+{
+    parser_.reset();
+}
+
+void PingParserExt::parseBuffer(const QByteArray& data)
+{
+    for(int i = 0; i < data.length(); i++) {
+        PingParser::ParseState state = parser_.parseByte(data.at(i));
+        if (state == PingParser::ParseState::NEW_MESSAGE) {
```

Comment:
Why ParseState is necessary ?

Reply:
It is to expose the current state of the parser. There are **three** possible states of interest: new message, parsing, and error.

Reply:
I'm saying that in some points you use `PingParser::NEW_MESSAGE`, `PingParser::ParseState::NEW_MESSAGE` and `Parser::ParserState::NEW_MESSAGE`.
There is two different classes and the enum name is not being used in some places resulting in 3 different enumeration uses. 
Is it possible to use the base class enum `Parser` with/without `ParserState` to help with readability and maintenance and avoiding the usage of a secondary kind ?